src/backend/utils/mmgr/aset.c

```diff
@@ -1117,6 +1118,8 @@ AllocSetDelete(MemoryContext context)
 		block = next;
 	}
 
+	MemoryContextNoteFree(&set->header, allFreed);
```

Comment:
I think if we believe that `context->allBytesAlloc - context->allBytesFreed` is equaled to the sum of sizes of allocated blocks we have to use this formula to exclude redundant computation in the body of loop.